Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/COREX12RealtimeTransforms.java

```diff
@@ -0,0 +1,387 @@
+import com.services.nhinc.schema.auditmessage.EventIdentificationType;
+import com.services.nhinc.schema.auditmessage.ParticipantObjectIdentificationType;
+import com.services.nhinc.schema.auditmessage.TypeValuePairType;
+import gov.hhs.fha.nhinc.common.auditlog.LogEventRequestType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
+import gov.hhs.fha.nhinc.common.nhinccommon.UserType;
+import gov.hhs.fha.nhinc.nhinclib.NullChecker;
+import gov.hhs.fha.nhinc.transform.audit.AuditDataTransformConstants;
+import gov.hhs.fha.nhinc.transform.audit.AuditDataTransformHelper;
+import gov.hhs.fha.nhinc.transform.audit.XDRTransforms;
+import gov.hhs.fha.nhinc.transform.marshallers.JAXBContextHandler;
+import java.io.ByteArrayOutputStream;
+import java.util.logging.Level;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import org.apache.log4j.Logger;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeRequest;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeResponse;
+
+/**
+ *
+ * @author svalluripalli
+ */
+public class COREX12RealtimeTransforms {
+
+    private static final Logger LOG = Logger.getLogger(COREX12RealtimeTransforms.class);
+    private XDRTransforms oXDR = new XDRTransforms();
+
+    /**
+     *
+     * @param msg
+     * @param assertion
+     * @param target
+     * @param direction
+     * @param _interface
+     * @return LogEventRequestType
+     */
+    public LogEventRequestType transformRequestToAuditMsg(COREEnvelopeRealTimeRequest request, AssertionType assertion, NhinTargetSystemType target, String direction, String _interface) {
+        LogEventRequestType result = null;
+        AuditMessageType auditMsg = null;
+
+        LOG.trace("Begin COREX12RealtimeTransforms -> transformRequestToAuditMsg() -- NHIN");
+        if (request == null) {
+            LOG.error("Requst Object was null");
+            return null;
+        }
+
+        if (assertion == null) {
+            LOG.error("Assertion was null");
+            return null;
+        }
+
+        // check to see that the required fields are not null
+        boolean missingReqFields = areRequiredCOREX12RealtimeFiledsNull(request, assertion);
+        if (missingReqFields) {
+            LOG.error("One or more required fields was missing");
+            return null;
+        }
+
+        result = new LogEventRequestType();
+        auditMsg = new AuditMessageType();
+
+        // Create EventIdentification
+        CodedValueType eventID = getCodedValueTypeForXDRResponse();
+
+        EventIdentificationType oEventIdentificationType = getEventIdentificationType(eventID);
+        auditMsg.setEventIdentification(oEventIdentificationType);
+
+        AuditMessageType.ActiveParticipant participant = getActiveParticipant(assertion.getUserInfo());
```

Comment:
ParticipantObjectIDTypeCode in ParticipantObjectIdentification should have code, codesystemname and displayname.